crates/cow-amm/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+mod event_updater;
+mod implementations;
+mod indexer;
+
+use {
+    crate::indexer::CowAmmRegistry,
+    ethcontract::{common::DeploymentInformation, Address},
+};
+pub use {event_updater::EventUpdater, implementations::safe_based::*, indexer::Indexer};
+
+pub trait CowAmm: Send + Sync {
+    /// Address of the CoW AMM.
+    /// Can be used by the autopilot to build the list of accepted cow amms.
+    fn address(&self) -> &Address;
+
+    /// Returns the list of tokens traded by this pool.
+    /// Can be used by the autopilot to build the list of native token prices to
+    /// query.
+    fn traded_tokens(&self) -> &[Address];
+}
+
+#[async_trait::async_trait]
+pub trait ContractHandler<E>: Send + Sync {
```

Comment:
If we can avoid it I think this type should not leak into the public.

Reply:
@MartinquaXD this particular trait has to be public because it has to be part of the restrictions for the `Registry` builder.